### ISSUE TITLE
Gwrite from Gdiff Blanks The Working Copy

### ISSUE THREAD

**Issue by raisin-loaf** [tpope/vim-fugitive#1102]
I figure I'm being an idiot here but I can't quite work out how.

From a `Gdiff`, if I run `Gwrite` from the index window the working copy window is entirely wiped. If I move to the working copy window and `undo` then I get the index version of the file and of course another `undo` brings back the working copy as it was at the start of the `Gdiff`.

From watching Drew Neil's excellent [video](http://vimcasts.org/episodes/fugitive-vim-working-with-the-git-index/) I was expecting `Gwrite` to overwrite the working copy with the version of the file in the index, resetting any changes that might have been made.

I get the same result running Vim with all other plugins disabled, reinforcing the notion that it's me that's at fault here. ;)

**Comment by tpope** [comment]
If you reload the other buffer instead of hitting undo, does it have the expected contents then? In other words, is the file correct on disk?

If you close the working tree buffer first, does `:Gwrite` work correctly then?

**Comment by raisin-loaf** [comment]
> If you reload the other buffer instead of hitting undo, does it have the expected contents then? In other words, is the file correct on disk?

No, it's blank on disk. To get the result I'm after (replacing the working copy with the index version) I have to switch to the working copy window, undo and then write.

> If you close the working tree buffer first, does :Gwrite work correctly then?

Yup, if I've got the working copy open in another tab and run :Gwrite from the index copy it's checked out correctly (and the working copy buffer is automatically refreshed).

On a side note, from the split window :Gdiff view running :Gread from the working copy is fine - the working copy buffer is updated with the index version. I could do it that way, but it always feels more logical for me to write from the index window. :)

Thanks for the speedy response!

**Comment by tpope** [comment]
What if you `:diffoff!` first?

**Comment by raisin-loaf** [comment]
> What if you `:diffoff!` first?

All is well, the working copy is replaced with the version in the index and the buffer updates.

**Comment by shxfee** [comment]
Im having the same issue. works with `:diffoff!` first.

**Comment by raisin-loaf** [comment]
Just in case it's helpful, I can't reproduce this issue using Neovim (v0.3.2-649-g512e0caae).

**Comment by tpope** [comment]
Based on the note in #859, I compiled Vim 8.0.101 and was able to reproduce the issue. Well, not exactly, at first glance the buffer looked empty, but actually the diff was out of sync. The full file was there if you scrolled down. I traced this to a Vim bug that was fixed in [8.0.421](https://github.com/vim/vim/commit/f58a8475e17bd566760fc7e2a17d35ddf4edacf2). Adding a `diffupdate` fixes the sync issue on versions prior to the patch.
